R/augsynth.R

```diff
@@ -103,12 +99,34 @@ fit_augsynth_internal <- function(wide, synth_data, Z, progfunc,
     }
     ## fit augsynth
     if(progfunc == "Ridge") {
+        # 1. find hyperparameter using CV call, if lambda is null
+        # 2. call fit_ridgeuag_formatted with best CV (if-else case below, depending on scm=T or F)
+        # TODO: remove if else statement here since scm is an arg
         if(scm) {
-            ## Ridge ASCM
-            augsynth <- do.call(fit_ridgeaug_formatted,
-                                list(wide_data = fit_wide, 
-                                   synth_data = fit_synth_data, 
-                                   Z = Z, V = V, ...))
+            # Ridge ASCM
+            if(is.null(list(...)$lambda)) {
+                lambda_results <- do.call(cv_ridge, list(wide_data = fit_wide, synth_data = fit_synth_data, Z = Z, progfunc = progfunc, 
+                                                         scm = scm, fixedeff = fixedeff, V = V, ...))
+                augsynth <- do.call(fit_ridgeaug_formatted,
+                                    list(wide_data = fit_wide,
+                                         synth_data = fit_synth_data,
+                                         Z = Z, V = V, ...))
+                augsynth$lambda <- lambda_results$lambda
+                augsynth$lambdas <- lambda_results$lambdas
+                augsynth$lambda_errors <- lambda_results$lambda_errors
+                augsynth$lambda_errors_se <- lambda_results$lambda_errors_se
+            } else {
+                augsynth <- do.call(fit_ridgeaug_formatted,
+                                    list(wide_data = fit_wide,
+                                         synth_data = fit_synth_data,
+                                         Z = Z, V = V, ...))
+            }
```

Comment:
I think there only needs to be one call to `fit_ridgeaug_formatted`, outside of the if else block. Something like

```suggestion
            if(is.null(list(...)$lambda)) {
                lambda_results <- do.call(cv_ridge, list(wide_data = fit_wide, synth_data = fit_synth_data, Z = Z, progfunc = progfunc, 
                                                         scm = scm, fixedeff = fixedeff, V = V, ...))
                lambda <- lambda_results$lambda
                lambdas <- lambda_results$lambdas
                lambda_errors <- lambda_results$lambda_errors
                lambda_errors_se <- lambda_results$lambda_errors_se
            } else {
                lambdas <- NULL
                lambda_errors <- NULL
                lambda_errors_se <- NULL
            }
            augsynth <- do.call(fit_ridgeaug_formatted,
                                    list(wide_data = fit_wide,
                                         synth_data = fit_synth_data,
                                         Z = Z, V = V, ...))
            augsynth$lambdas <- lambdas
            augsynth$lambda_errors <- lambda_errors
            augsynth$lambda_errors_se <- lambda_errors_se
```

Reply:
Added. :)